1159/en.md

```diff
@@ -0,0 +1,82 @@
+
+# [LOJ 1159 Batman](https://lightoj.com/problem/batman)
+
+## Solution Approach:
+In this problem given 3 string and need to find the length of the longest common subsequence 
+[LCS](https://www.programiz.com/dsa/longest-common-subsequence) of these three strings.
+
+Its not necessary that the characters have to be next to each other 
+in the original strings.
+
+One approach to solve the problem is to create a 3-dimensional table with dimensions corresponding
+to the lengths of each of the three input strings.
+The sub-problems in this case are the substrings 
+dp[1...i], Y[1...j], Z[1...k] for all possible values of i, j, k
+
+To compute the value of LCS[i][j][k], we consider three possibilities:
+- You can start by checking base case whether the length of any string is 0, in which case the LCS is 0.
+- if three characters match, length of the common subsequence would be 1 plus 
+  the length of the common subsequence till the i-1,j-1 and k-1 indexes
+- Otherwise, the LCS is will be maximum of value of the element above or to the left of the 
+  current element of three strings(max of neighbour elements).
```

Comment:
I would rephrase it. Something like:

> Otherwise, we can drop the last character of any one of the three strings and take the maximum value i.e. `LCS[i][j][k] = max(LCS[i-1][j][k], LCS[i][j-1][k], LCS[i][j][k-1])`.